File: app/src/main/java/com/kunfei/bookshelf/widget/page/TxtPage.java
package com.kunfei.bookshelf.widget.page;

import java.util.ArrayList;
import java.util.List;

/**
 * 页面
 */

class TxtPage {
    private int position;
    private String title;
    private int titleLines; //当前 lines 中为 title 的行数。
    private List<String> lines = new ArrayList<>();

    TxtPage(int position) {
        this.position = position;
    }

    int getPosition() {
        return position;
    }

    String getTitle() {
        return title;
    }

    void setTitle(String title) {
        this.title = title;
    }

    int getTitleLines() {
        return titleLines;
    }

    void setTitleLines(int titleLines) {
        this.titleLines = titleLines;
    }

    String getContent() {
        StringBuilder s = new StringBuilder();
        if (lines != null) {
            for (int i = 0; i < lines.size(); i++) {
                s.append(lines.get(i));
            }
        }
        return s.toString();
    }

    void addLine(String line) {
        lines.add(line);
    }

    void addLines(List<String> lines) {
        this.lines.addAll(lines);
    }

    String getLine(int i) {
        return lines.get(i);
    }

    List<String> getLines() {
        return lines;
    }

    int size() {
        return lines.size();
    }
}

File: app/src/main/java/com/kunfei/bookshelf/widget/page/TxtPage.kt
package com.kunfei.bookshelf.widget.page

import java.util.*

/**
 * 页面
 */

class TxtPage(val position: Int) {
    var title: String? = null
    var titleLines: Int = 0 //当前 lines 中为 title 的行数。
    private val lines = ArrayList<String>()
    //存放每个字的位置
    var txtLists: List<TxtLine>? = null

    val content: String
        get() {
            val s = StringBuilder()
            for (i in lines.indices) {
                s.append(lines[i])
            }
            return s.toString()
        }

    fun addLine(line: String) {
        lines.add(line)
    }

    fun addLines(lines: List<String>) {
        this.lines.addAll(lines)
    }

    fun getLine(i: Int): String {
        return lines[i]
    }

    fun getLines(): List<String> {
        return lines
    }

    fun size(): Int {
        return lines.size
    }
}
